From 309bbdefcf2c93dfa1ae42ad9b06c37c77584c4c Mon Sep 17 00:00:00 2001
From: Prasanth Jayachandran <prasanthj@apache.org>
Date: Thu, 2 Mar 2017 20:49:26 -0800
Subject: [PATCH 1004/1164] CDH-51070 HIVE-16086: Fix
 HiveMetaStoreChecker.checkPartitionDirsSingleThreaded
 method (Kiran Kumar Kolli reviewed by Prasanth
 Jayachandran)

(cherry picked from commit b62246540d604bbb0343e2dc1cec7411e60eacd9)

Change-Id: I0806c03d535b6ff1ce3336549782141639d6b6e3
---
 .../hive/ql/metadata/HiveMetaStoreChecker.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveMetaStoreChecker.java b/ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveMetaStoreChecker.java
index 8eb8154..8d41197 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveMetaStoreChecker.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveMetaStoreChecker.java
@@ -535,9 +535,9 @@ private void checkPartitionDirsInParallel(final ThreadPoolExecutor pool,
    */
   private void checkPartitionDirsSingleThreaded(Queue<Path> basePaths, final Set<Path> allDirs,
       final FileSystem fs, final int depth, final int maxDepth) throws IOException, HiveException {
-    final Queue<Path> nextLevel = new LinkedList<>();
     for (final Path path : basePaths) {
       FileStatus[] statuses = fs.listStatus(path, FileUtils.HIDDEN_FILES_PATH_FILTER);
+      final Queue<Path> nextLevel = new LinkedList<>();
       boolean fileFound = false;
       for (FileStatus status : statuses) {
         if (status.isDirectory()) {
-- 
1.7.9.5

